pkg/synthetictests/duplicated_events.go

```diff
@@ -389,86 +411,106 @@ func getTimesAnEventHappened(message string) (string, int) {
 	return matches[0][1], int(times)
 }
 
-// isConsoleReadinessDuringInstallation returns true if the event is for console readiness and it happens during the
-// initial installation of the cluster.
-// we're looking for something like
-// > ns/openshift-console pod/console-7c6f797fd9-5m94j node/ip-10-0-158-106.us-west-2.compute.internal - reason/ProbeError Readiness probe error: Get "https://10.129.0.49:8443/health": dial tcp 10.129.0.49:8443: connect: connection refused
-// with a firstTimestamp before the cluster completed the initial installation
-func isConsoleReadinessDuringInstallation(monitorEvent monitorapi.EventInterval, kubeClientConfig *rest.Config) bool {
-	if !strings.Contains(monitorEvent.Locator, "ns/openshift-console") {
-		return false
-	}
-	if !strings.Contains(monitorEvent.Locator, "pod/console-") {
-		return false
-	}
-	if !strings.Contains(monitorEvent.Locator, "Readiness probe") {
-		return false
-	}
-	if !strings.Contains(monitorEvent.Locator, "connect: connection refused") {
-		return false
-	}
-	tokens := strings.Split(monitorEvent.Locator, " ")
-	tokens = strings.Split(tokens[1], "/")
-	podName := tokens[1]
-
-	if kubeClientConfig == nil {
-		// default to OK
-		return true
-	}
-
-	// This block gets the time when the cluster completed installation
+func getInstallCompletionTime(kubeClientConfig *rest.Config) *metav1.Time {
 	configClient, err := configclient.NewForConfig(kubeClientConfig)
 	if err != nil {
-		// default to OK
-		return true
+		return nil
 	}
 	clusterVersion, err := configClient.ConfigV1().ClusterVersions().Get(context.TODO(), "version", metav1.GetOptions{})
 	if err != nil {
-		// default to OK
-		return true
+		return nil
 	}
 	if len(clusterVersion.Status.History) == 0 {
-		// default to OK
-		return true
+		return nil
+	}
+	return clusterVersion.Status.History[len(clusterVersion.Status.History)-1].CompletionTime
+}
+
+func getMatchedElementsFromMonitorEventMsg(regExp *regexp.Regexp, message string) (error, string, string, string, string, string) {
+	var namespace, pod, node, reason, msg string
+	if !regExp.MatchString(message) {
+		return errors.New("regex match error"), namespace, pod, node, reason, msg
+	}
+	subMatches := regExp.FindStringSubmatch(message)
+	subNames := regExp.SubexpNames()
+	for i, name := range subNames {
+		switch name {
+		case "NS":
+			namespace = subMatches[i]
+		case "POD":
+			pod = subMatches[i]
+		case "NODE":
+			node = subMatches[i]
+		case "REASON":
+			reason = subMatches[i]
+		case "MSG":
+			msg = subMatches[i]
+		}
 	}
-	initialInstallHistory := clusterVersion.Status.History[len(clusterVersion.Status.History)-1]
-	if initialInstallHistory.CompletionTime == nil {
+	return nil, namespace, pod, node, reason, msg
```

Comment:
if any of these are empty, we should fail, right?  Seems like it means the regex failed.